### PR TITLE
Review Branch - Gilmour Proxy

### PR DESCRIPTION
**Pull request by poorva29**

### CODE REVIEW

File: tests/gilmour_proxy_test.go [hunk 1]
```diff
@@ -0,0 +1,304 @@
+package tests
```
Comment (adityagodbole) [inline tests/gilmour_proxy_test.go]: If you are testing the proxy package, the tests should also be in the proxy package. Also, the test file should be called `<filename>_test.go`

File: proxy/proxy.go [hunk 2]
```diff
@@ -0,0 +1,506 @@
...
+	}
+	if err = node.AddSlots(node.slots); err != nil {
+		return
+	}
+	return
+}
+
+func CreateNode(nodeReq *NodeReq) (node *Node, err error) {
+	node = new(Node)
+	engine, _ := makeGilmour("127.0.0.1:6379")
```
Comment (adityagodbole) [inline proxy/proxy.go]: This cannot be hardcoded to localhost. The `CreateNode` function should instead take a `Gilmour` object in it's params

Comment (poorva29) [inline proxy/proxy.go]: The hardcoding will change once we have a config file or some mode by which user can interact
I will pass Gilmour object in params for `CreateNode`

File: proxy/proxy.go [hunk 3]
```diff
@@ -0,0 +1,705 @@
...
+}
+
+// GetNodeMap returns nodeMap
+func GetNodeMap() nodeMap {
+	return *nMap
+}
+
+// Put adds node in nodeMap
+func (n *nodeMap) Put(uid NodeID, node *Node) (err error) {
+	n.Mutex.Lock()
```
Comment (meson10) [inline proxy/proxy.go:81]: Good practice is to have `defer n.Mutex.Unlock()` right after you acquire a lock.

Comment (meson10) [inline proxy/proxy.go:81]: Good practice is to have `defer n.Mutex.Unlock()` right after you acquire a lock.

File: proxy/proxy.go [hunk 4]
```diff
@@ -0,0 +1,705 @@
...
+
+// NodeID is a string to hold node's id
+type NodeID string
+
+// GilmourTopic is a string to hold topic for signal or request
+type GilmourTopic string
+
+// implements NodeMapOperations
+type nodeMap struct {
+	sync.Mutex
```
Comment (meson10) [inline proxy/proxy.go:53]: Shouldn't you use a RWMutex instead of a Mutex? That way Read Locks can be distinguished from a write Lock.

**Pull request merged**